src/v/cluster/topics_frontend.cc

```diff
@@ -622,7 +622,37 @@ ss::future<std::vector<topic_result>> topics_frontend::delete_topics(
 
 ss::future<topic_result> topics_frontend::do_delete_topic(
   model::topic_namespace tp_ns, model::timeout_clock::time_point timeout) {
-    delete_topic_cmd cmd(tp_ns, tp_ns);
+    // Look up config
+    auto topic_meta_opt = _topics.local().get_topic_metadata_ref(tp_ns);
+    if (!topic_meta_opt.has_value()) {
+        topic_result result(std::move(tp_ns), errc::topic_not_exists);
+        return ss::make_ready_future<topic_result>(result);
+    }
+    auto& topic_meta = topic_meta_opt.value().get();
+
+    // Default to traditional deletion, without tombstones
+    // Use tombstones for tiered storage topics that require remote erase
+    topic_lifecycle_transition_mode mode
+      = topic_meta.get_configuration().properties.requires_remote_erase()
+          ? topic_lifecycle_transition_mode::pending_gc
+          : topic_lifecycle_transition_mode::oneshot_delete;
```

Comment:
`segments_outside_manifest` will still be normal for topics that haven't been deleted yet, and for the period between deleting a topic and the scrubber getting around to cleaning up, so it'll remain non-fatal in the tool.

However, in the next phase of changes that add cleanup of orphans, I intend to change the post-test routine to do one scrub, then delete topics, then wait for a scrub, then finally change the `permitted_anomalies` in redpanda.py to stop permitting these errors, so that we'll have a totally deterministic end state for the tests, and anything going wrong with the deletion/cleanup.  However probably not on every test because it'll bulk out the runtime.